src/main/java/com/google/maps/StaticMapsRequest.java

```diff
@@ -0,0 +1,415 @@
+    }
+
+    @Override
+    public String toUrlValue() {
+      return format;
+    }
+  }
+
+  /**
+   * <code>format</code> defines the format of the resulting image. By default, the Google Static
+   * Maps API creates PNG images. There are several possible formats including GIF, JPEG and PNG
+   * types.
+   *
+   * @param format The format of the static map.
+   * @return Returns this {@code StaticMapsRequest} for call chaining.
+   */
+  public StaticMapsRequest format(ImageFormat format) {
+    return param("format", format);
+  }
+
+  public enum StaticMapType implements UrlValue {
+    roadmap,
+    satellite,
+    terrain,
+    hybrid;
+
+    public String toUrlValue() {
+      return this.name();
+    }
+  }
+
+  /**
+   * <code>maptype</code> defines the type of map to construct.
+   *
+   * @param maptype The map type of the static map.
+   * @return Returns this {@code StaticMapsRequest} for call chaining.
+   */
+  public StaticMapsRequest maptype(StaticMapType maptype) {
+    return param("maptype", maptype);
+  }
+
+  /**
+   * <code>region</code> defines the appropriate borders to display, based on geo-political
+   * sensitivities. Accepts a region code specified as a two-character ccTLD ('top-level domain')
+   * value.
+   *
+   * @param region The region of the static map.
+   * @return Returns this {@code StaticMapsRequest} for call chaining.
+   */
+  public StaticMapsRequest region(String region) {
+    return param("region", region);
+  }
+
+  private static final Pattern colorPattern =
+      Pattern.compile(
+          "(0x\\p{XDigit}{6})|(black)|(brown)|(green)|(purple)|(yellow)|(blue)|(gray)|(orange)|(red)|(white)");
+
+  public static class Markers implements UrlValue {
+
+    public static enum MarkersSize implements UrlValue {
+      tiny,
+      mid,
+      small,
+      normal;
+
+      @Override
+      public String toUrlValue() {
+        return this.name();
+      }
+    }
+
+    public static enum CustomIconAnchor implements UrlValue {
+      top,
+      bottom,
+      left,
+      right,
+      center,
+      topleft,
+      topright,
+      bottomleft,
+      bottomright;
+
+      @Override
+      public String toUrlValue() {
+        return this.name();
+      }
+    }
+
+    private MarkersSize size;
+    private String color;
+    private String label;
+    private String customIconURL;
+    private CustomIconAnchor anchorPoint;
+    private List<String> locations = new ArrayList<>();
+
+    /**
+     * Specifies the size of marker. If no size parameter is set, the marker will appear in its
+     * default (normal) size.
+     */
+    public void size(MarkersSize size) {
+      this.size = size;
+    }
+
+    /**
+     * Specifies a 24-bit color (example: color=0xFFFFCC) or a predefined color from the set {black,
+     * brown, green, purple, yellow, blue, gray, orange, red, white}.
+     */
+    public void color(String color) {
+      if (!colorPattern.matcher(color).matches()) {
```

Comment:
is it worth enforcing this locally? We might allow more colors in future.

Reply:
Fair call, ripped out regex